### ISSUE TITLE
Missing column oidc_id_token_claims at schema oauth2-authorization-schema.sql

### ISSUE THREAD

**Issue by filipemb** [spring-projects/spring-authorization-server#1638]
**Describe the bug**
If user follow guide: [How-to: Implement core services with JPA](https://docs.spring.io/spring-authorization-server/reference/guides/how-to-jpa.html) will face an exception during authentication because [Authorization.java](https://github.com/spring-projects/spring-authorization-server/blob/0.4.x/docs/src/docs/asciidoc/examples/src/main/java/sample/jpa/entity/authorization/Authorization.java)
has a map to oidcIdTokenClaims

```
@Column(length = 2000)
private String oidcIdTokenClaims;

```
that is not present at table dll script.

It misses:
```
ALTER TABLE oauth2_authorization
    ADD oidc_id_token_claims blob DEFAULT NULL;
```

**To Reproduce**
1) follow steps from guide: [How-to: Implement core services with JPA](https://docs.spring.io/spring-authorization-server/reference/guides/how-to-jpa.html)
2) start authorization server and login in.

**Expected behavior**
Login happens without failure after replacing:
JdbcOAuth2AuthorizationService, JdbcRegisteredClientRepository and JdbcOAuth2AuthorizationConsentService 
by 
JpaOAuth2AuthorizationService, JpaRegisteredClientRepository and JpaOAuth2AuthorizationConsentService

**Sample**
I'll fail reporting this (1 AM here in Brazil, sry guys! :wink: )

Reports that include a sample will take priority over reports that do not.
At times, we may require a sample, so it is good to try and include a sample up front.

**Comment by jgrandja** [comment]
@filipemb I'm not seeing any issue here. Take a look at [JpaTests](https://github.com/spring-projects/spring-authorization-server/blob/25b70bcafe2034caa17eea53c424aaf5754225b3/docs/src/test/java/sample/jpa/JpaTests.java#L90) as it passes using all the JPA components.

**Comment by filipemb** [comment]
Hi @jgrandja, 
Analyzing the test you are right:
Test [JpaTests](https://github.com/spring-projects/spring-authorization-server/blob/25b70bcafe2034caa17eea53c424aaf5754225b3/docs/src/test/java/sample/jpa/JpaTests.java#L90) at method `oidcLoginWhenJpaCoreServicesAutowiredThenUsed() ` uses
`OAuth2Authorization authorization = findAuthorization(accessToken, OAuth2ParameterNames.ACCESS_TOKEN);`
to interact with mapped [Authorization.java](https://github.com/spring-projects/spring-authorization-server/blob/0.4.x/docs/src/docs/asciidoc/examples/src/main/java/sample/jpa/entity/authorization/Authorization.java),
and covers method `findAuthorization `from `JpaOAuth2AuthorizationService` 
this method will return an `OAuth2Authorization` after querying by token type using `AuthorizationRepository`.
The conversion from Authorization to OAuth2Authorization will only deal with the field **oidcIdTokenClaims** if the retrieved entity has an oidcIdTokenValue.
The test expects that the retrieved authorization has it as it asserts:
`assertThat(authorization.getToken(idToken)).isNotNull();`
A token with a non null oidcIdTokenValue will be of type "token_type": "id token". 

Investigating further I found my issue, it was caused by SQL scripts from an old guide changed by @sjohnr long ago at this
[commit](https://github.com/spring-projects/spring-authorization-server/commit/7edbe2a30dacf3bcf7cd3466a6d802e20dd243b3#diff-5cf17d90e817709ef37afa5d4c10ad561158a2cf60769750cb3e2bd5be3b4178R95)
and not totally covered by Wiki [Spring-Authorization-Server-1.1-Migration-Guide](https://github.com/spring-projects/spring-authorization-server/wiki/Spring-Authorization-Server-1.1-Migration-Guide)

So my suggestion is to fix this script:
[oauth2-authorization-schema.sql](https://github.com/spring-projects/spring-authorization-server/blob/main/oauth2-authorization-server/src/main/resources/org/springframework/security/oauth2/server/authorization/oauth2-authorization-schema.sql
)

to reflect the same from the [How to JPA Guide](https://github.com/spring-projects/spring-authorization-server/blob/main/docs/modules/ROOT/pages/guides/how-to-jpa.adoc)
and fix Wiki [Spring-Authorization-Server-1.1-Migration-Guide](https://github.com/spring-projects/spring-authorization-server/wiki/Spring-Authorization-Server-1.1-Migration-Guide)

Tkx for your attention on this topic.
Wish u the best. :muscle:
From Brazil :beginner:

**Comment by jgrandja** [comment]
@filipemb 

> So my suggestion is to fix this script:
[oauth2-authorization-schema.sql](https://github.com/spring-projects/spring-authorization-server/blob/main/oauth2-authorization-server/src/main/resources/org/springframework/security/oauth2/server/authorization/oauth2-authorization-schema.sql)
>
> to reflect the same from the [How to JPA Guide](https://github.com/spring-projects/spring-authorization-server/blob/main/docs/modules/ROOT/pages/guides/how-to-jpa.adoc)

The `oauth2-authorization-schema.sql` is specifically defined for `JdbcOAuth2AuthorizationService`. Whereas, the 
[JPA authorization schema](https://docs.spring.io/spring-authorization-server/reference/guides/how-to-jpa.html#authorization-schema) is designed for `JpaOAuth2AuthorizationService`. These are 2 different implementations of `OAuth2AuthorizationService`, which have their own (and different) schema definitions.

I'm going to close this as I don't see any issues.

If you still think there is a bug please provide a [minimal sample](https://stackoverflow.com/help/minimal-reproducible-example) that reproduces the issue so I can look into it further.
